### PR TITLE
[git-webkit] Automatically grant team access to private forks

### PR DESCRIPTION
**Pull request by JonWBedard**
#### 979212d051b649f3b48adcb9d83b5ec621fa0116
<pre>
[git-webkit] Automatically grant team access to private forks
<a href="https://bugs.webkit.org/show_bug.cgi?id=240819">https://bugs.webkit.org/show_bug.cgi?id=240819</a>
&lt;rdar://93780380 &gt;

Reviewed by Stephanie Lewis.

* Tools/Scripts/libraries/webkitbugspy/webkitbugspy/github.py:
(Tracker): Share ACCEPT_HEADER string.
(Tracker.credentials.validater): Ditto.
* Tools/Scripts/libraries/webkitscmpy/webkitscmpy/program/setup.py:
(Setup.github): Allow provided team access to forked repository.
(Setup.git): Pass team for repo access from webkitscmpy.access config value.
* Tools/Scripts/libraries/webkitscmpy/webkitscmpy/remote/git_hub.py:
(GitHub): Share ACCEPT_HEADER string.
(GitHub.PRGenerator.create): Ditto.
(GitHub.PRGenerator.update): Ditto.
(GitHub.request): Ditto.
* metadata/git_config_extension: Define group to have access to apple/WebKit forks.

Canonical link: <a href="https://commits.webkit.org/251545@main">https://commits.webkit.org/251545@main</a>
git-svn-id: <a href="https://svn.webkit.org/repository/webkit/trunk@295540">https://svn.webkit.org/repository/webkit/trunk@295540</a> 268f45cc-cd09-0410-ab3c-d52691b4dbfc
</pre>

### CODE REVIEW

File: Tools/Scripts/libraries/webkitscmpy/webkitscmpy/program/setup.py [hunk 1]
```diff
@@ -47,16 +47,54 @@ def github(cls, args, repository, additional_setup=None, remote=None, **kwargs):
         if additional_setup:
             result += additional_setup(args, repository)
 
+        team_id = None
+        auth = HTTPBasicAuth(username, access_token)
```
Comment (sundiamonde) [inline Tools/Scripts/libraries/webkitscmpy/webkitscmpy/program/setup.py:51]: seems like we should do a check to see if username is a member of the team?

**Comment by sundiamonde** [comment]
Why should I be concerned that webkitcmppy is changing my changelog resolution?  Changing it how?  If this is a problem maybe the user should be prompted, it its not maybe warning is the wrong tag?

**Comment by JonWBedard** [comment]
> Why should I be concerned that webkitcmppy is changing my changelog resolution? Changing it how? If this is a problem maybe the user should be prompted, it its not maybe warning is the wrong tag?

Warning _is_ the wrong tag, we need to implement custom logging handlers in `webkitcorepy`. The only reason we haven't done this is priority. Logging streams are controlled by `-v` and `-q`, in this specific case, we want to let the user know we're doing something by default (we're setting up a script to merge conflicted changelogs), but we also want to tie that message to `-q` so it can be suppressed. (unlike, for example, something printed to stdout or stderr).

That being said, we should also just remove the whole conflicted changelog thing since ChangeLogs are dead.

File: Tools/Scripts/libraries/webkitscmpy/webkitscmpy/program/setup.py (hunk 1 above)
Comment (sundiamonde) [reply Tools/Scripts/libraries/webkitscmpy/webkitscmpy/program/setup.py:51]: ^

**Review by sundiamonde — APPROVED** [review]

File: Tools/Scripts/libraries/webkitscmpy/webkitscmpy/program/setup.py (hunk 1 above)
Comment (JonWBedard) [reply Tools/Scripts/libraries/webkitscmpy/webkitscmpy/program/setup.py:51]: Oh, I see your point, not just checking if the user has access to the team, but verify they're part of it. Will make that change.

**Pull request merged**